### PR TITLE
[IMP/REF] test_lint, *: enable some linting rules

### PR DESCRIPTION
**Pull request by jpp-odoo**
The commits enable some rules. Some work was done to allow some of these rules to be enabled, and offending code was adapted.

### CODE REVIEW

File: addons/web/static/src/legacy/js/views/kanban/kanban_model.js [hunk 1]
```diff
@@ -402,7 +402,7 @@ var KanbanModel = BasicModel.extend({
     async _readProgressBarGroup(list, options) {
         const groupsDef = this._readGroup(list, options);
         const progressBarDef = this._readProgressBar(list);
-        const [groups, progressBar] = await Promise.all([groupsDef, progressBarDef]);
+        const [_groups, progressBar] = await Promise.all([groupsDef, progressBarDef]);
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/js/views/kanban/kanban_model.js:405]:
```js
 const [/* groups */, progressBar]
```

**Comment by xmo-odoo** [comment]
The `caughtError` stuff seems non-idiomatic in general, in JS, and for the project specifically: pretty much everywhere the way to say "I know I've something here but I don't care" is to use `_` as a prefix for the variable (or as its name, though that can have slightly different semantics depending on the language).

(also ES2019 makes catch bindings optional, maybe that's an option? I'm not entirely clear what the configuration file implies by having an `env.es2017: true` but an `ecmaVersion: 2019`)

**Comment by sdegueldre** [comment]
> (also ES2019 makes catch bindings optional, maybe that's an option? I'm not entirely clear what the configuration file implies by having an `env.es2017: true` but an `ecmaVersion: 2019`)

env is for which global objects will be present at runtime (eg, Promise, Map, Set, ...) while ecmaVersion is is for syntax features which impacts the parser (while `new Map()` will parse, it's not valid without env: es>2015, an async function will not parse without a sufficient ecmaVersion). Depending on what particular combo of transpilers and polyfills you're using, having a higher env version or a higher ecmaVersion are both valid. For our use case we can probably bump the env to es2019 since we're not using either polyfills or transpiling syntax features.

We're not even close to having made a decision on ignore patterns, whether for errors or variable, you'll probably be pinged as needed when the PR is ready, but I think it's unlikely we keep the current "ignore" prefix for errors.

File: addons/mail/static/src/models/rtc/rtc.js [hunk 2]
```diff
@@ -288,7 +288,7 @@ registerModel({
                 try {
                     const audioStream = await browser.navigator.mediaDevices.getUserMedia({ audio: this.messaging.userSetting.getAudioConstraints() });
                     audioTrack = audioStream.getAudioTracks()[0];
-                } catch (e) {
+                } catch (_e) {
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:291]: This looks like noise, make them empty `catch` instead?

**Review by seb-odoo — COMMENTED** [review]
Thank you for the heads up, and good clean up.

If you remove variables that are indeed unused then it's obviously ok.

The minimal diff would be `var time = require('web.time');` -> `require('web.time');`
If you go further than that and remove the line, make sure the `import`/`require` that you remove doesn't have side effects that would be lost by not calling it (I'm just mentioning it in general, I'm not aware of one where it would be problematic).

**Comment by ged-odoo** [comment]
robodoo r+

File: addons/mail/static/src/models/rtc/rtc.js (hunk 2 above)
Comment (xmo-odoo) [reply addons/mail/static/src/models/rtc/rtc.js:291]: That's what I ask about above, but the JS team is understandably somewhat wary of making such a change as it's not that old a feature (it's ES2019).

**Comment by jpp-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**